### PR TITLE
Simplify Iter/IntoIter implementations

### PR DESCRIPTION
**Pull request by GnomedDev**

### CODE REVIEW

File: src/into_iterator_impl.rs [hunk 1]
```diff
@@ -12,22 +10,17 @@ impl<T, const CAP: usize> IntoIterator for OptArrayVec<T, CAP> {
...
 	fn next(&mut self) -> Option<Self::Item> {
-		if self.index < self.len {
-			let ret = self.parent.inner[self.index].take();
-			self.index += 1;
-			ret
-		} else {
-			None
+		match self.inner.next() {
+			Some(Some(item)) => Some(item),
+			Some(None) | None => None,
```
Comment (FlareFlo) [inline src/into_iterator_impl.rs:24]: could use `flatten` alternatively?

File: src/into_iterator_impl.rs [hunk 2]
```diff
@@ -1,9 +1,7 @@
 use crate::OptArrayVec;
 
 pub struct IntoIter<T, const CAP: usize> {
-	parent: OptArrayVec<T, CAP>,
-	index: usize,
-	len: usize,
+	inner: std::array::IntoIter<Option<T>, CAP>,
```
Comment (FlareFlo) [inline src/into_iterator_impl.rs:4]: using std would mean we lose possible no_std capability by default, maybe try core::*?

File: src/into_iterator_impl.rs (hunk 1 above)
Comment (GnomedDev) [reply src/into_iterator_impl.rs:24]: Flatten only works for `Option<Option<T>>` -> `Option<T>`, this is `Option<&Option<T>>`.

File: src/into_iterator_impl.rs (hunk 2 above)
Comment (GnomedDev) [reply src/into_iterator_impl.rs:4]: This library is currently not no_std, as it lacks the `no_std` attribute at crate level, although I can submit a PR for that if you want.

Comment (FlareFlo) [reply src/into_iterator_impl.rs:4]: i can imagine it would be useful as i don't see any dependence on std for now

**Pull request merged**